src/base/Roles/TellerWithMultiAssetSupport.sol

```diff
@@ -530,6 +547,9 @@ contract TellerWithMultiAssetSupport is Auth, BeforeTransferHook, ReentrancyGuar
         shares = depositAmount.mulDivDown(ONE_SHARE, accountant.getRateInQuoteSafe(depositAsset));
         shares = asset.sharePremium > 0 ? shares.mulDivDown(1e4 - asset.sharePremium, 1e4) : shares;
         if (shares < minimumMint) revert TellerWithMultiAssetSupport__MinimumMintNotMet();
+        if (depositCap != type(uint96).max) {
```

Comment:
You could save depositCap in a memory variable here to save another SLOAD(100 gas)